fe1-web/parts/lao/socialMedia/SocialHome.tsx

```diff
@@ -45,12 +48,38 @@ const styles = StyleSheet.create({
 const SocialHome = () => {
   const [inputChirp, setInputChirp] = useState('');
 
+  const laoSelect = makeCurrentLao();
+  const lao = useSelector(laoSelect);
+  let userPublicKey: PublicKey | undefined;
+
+  if (lao === undefined) {
+    throw new Error('LAO is currently undefined, impossible to access to Social Media');
+  }
+
+  // Get the pop token of the user using the last tokenized roll call
+  const rollCallId = lao.last_tokenized_roll_call_id;
+  const eventSelect = makeEventGetter(lao.id, rollCallId);
+  const rollCall: RollCall = useSelector(eventSelect) as RollCall;
+
+  // This will be run again each time the lao.last_tokenized_roll_call_id changes
+  useEffect(() => {
+    generateToken(lao.id, rollCallId).then((token) => {
+      if (token && rollCall.containsToken(token)) {
+        userPublicKey = token.publicKey;
+      }
+    });
+  }, [lao.last_tokenized_roll_call_id]);
+
   const publishChirp = () => {
-    requestAddChirp(inputChirp)
-      .catch((err) => {
-        console.error('Could not add chirp, error:', err);
-      });
-    setInputChirp('');
+    if (userPublicKey) {
```

Comment:
This will fail until useEffect + generateToken runs.
It'd be more robust to disable publishing until userPublicKey is set.
I also suspect that the userPublicKey should be a state for this to work correctly 🤔

Reply:
I did it !